src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -103,6 +135,33 @@ public void Add(X509Certificate2 certificate)
             _storePal.Add(certificate.Pal);
         }
 
+        public void AddRange(X509Certificate2Collection certificates)
+        {
+            if (certificates == null)
+                throw new ArgumentNullException(nameof(certificates));
+
+            int i = 0;
+            try
+            {
+                foreach (X509Certificate2 certificate in certificates)
+                {
+                    Add(certificate);
+                    i++;
+                }
+            }
+            catch
+            {
+                // For desktop compat, we keep the exception semantics even though they are not ideal
+                // because a duplicate will raise exception causing certs to be removed even if 
+                // they weren't there before.
+                for (int j = 0; j < i; j++)
+                {
+                    Remove(certificates[j]);
```

Comment:
Yes the collection can be corrupted during exception; that's why I added the comment. Besides the comment, the method is identical to netfx. I will update the comments for the exception semantics -- it does not occur as a result of duplicate or missing (duplicate replaces; missing ignores) however there are many reasons an exception can be thrown for example if the NotBefore or NotAfter is wrong (https://msdn.microsoft.com/en-us/library/windows/desktop/aa376009(v=vs.85).aspx) or if the store is readonly.